nixos/modules/services/web-apps/chasemapper.nix

```diff
@@ -0,0 +1,492 @@
+          car_source_port = lib.mkOption {
+            default = 55672;
+            description = ''
+              Car position source port (UDP)
+            '';
+            type = lib.types.int;
+          };
+          online_tracker = lib.mkOption {
+            default = "sondehubamateur";
+            description = ''
+              Online Tracker System
+            '';
+            type = lib.types.str;
+          };
+        };
+
+        gpsd = {
+          gpsd_host = lib.mkOption {
+            default = "localhost";
+            description = ''
+              GPSD Host
+            '';
+            type = lib.types.str;
+          };
+          gpsd_port = lib.mkOption {
+            default = 2947;
+            description = ''
+              GPSD Port
+            '';
+            type = lib.types.int;
+          };
+        };
+
+        gps_serial = {
+          gps_port = lib.mkOption {
+            default = "/dev/ttyUSB0";
+            description = ''
+              GPS serial device
+            '';
+            type = lib.types.str;
+          };
+          gps_baud = lib.mkOption {
+            default = 9600;
+            description = ''
+              GPS baud rate
+            '';
+            type = lib.types.int;
+          };
+        };
+
+        map = {
+          flask_host = lib.mkOption {
+            default = "0.0.0.0";
+            description = ''
+              Host to host webserver on
+            '';
+            type = lib.types.str;
+          };
+          flask_port = lib.mkOption {
+            default = 5001;
+            description = ''
+              Port for web server to listen on
+            '';
+            type = lib.types.int;
+          };
+          default_lat = lib.mkOption {
+            default = -34.9;
+            description = ''
+              Default Map Latitude
+            '';
+            type = lib.types.float;
+          };
+          default_lon = lib.mkOption {
+            default = 138.6;
+            description = ''
+              Default Map Longitude
+            '';
+            type = lib.types.float;
+          };
+          default_alt = lib.mkOption {
+            default = 0.0;
+            description = ''
+              Default Map Altitude
+            '';
+            type = lib.types.float;
+          };
+          payload_max_age = lib.mkOption {
+            default = 180;
+            description = ''
+              How long to keep payload data (minutes)
+            '';
+            type = lib.types.int;
+          };
+          thunderforest_api_key = lib.mkOption {
+            default = "none";
+            description = ''
+              ThunderForest API Key
+            '';
+            type = lib.types.str;
+          };
+          stadia_api_key = lib.mkOption {
+            default = "none";
+            description = ''
+              Stadia Maps API Key
+            '';
+            type = lib.types.str;
+          };
```

Comment:
Are these api keys secrets? If so, please consider using `api_key_file` since `/nix/store` is a public place.

Reply:
Isn't this limited by chasemapper itself? It has no way to read in an API key file afaik, and this configuration is just used to generate chasemapper's config file (which ends up in /nix/store). So I would think there's not a ton we can do about this